infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientRouter.java

```diff
@@ -144,11 +144,15 @@ public IClientRouter addClient(IClient client) {
         client.setRouter(this);
 
         // Iterate through all types of CorfuMsgType, registering the handler
-        client.getHandledTypes().stream()
-                .forEach(x -> {
-                    handlerMap.put(x, client);
-                    log.info("Registered client to handle messages of type {}", x);
-                });
+        try {
+            client.getHandledTypes().stream()
+                    .forEach(x -> {
+                        handlerMap.put(x, client);
+                        log.info("Registered client to handle messages of type {}", x);
+                    });
+        } catch (Exception ex) {
+            // No Legacy getHandledTypes - Ignore
```

Comment:
Why can't we explicitly fail this, or log?

Also, what if the thread was interrupted, swallowing all exceptions might cause problems, like "hanging".

Reply:
We don't want to fail the creation of the router just because a particular client (or server in the case of a server-side router) doesn't implement a legacy message handler. If it doesn't, we simply want to continue on since it isn't a sign that something has gone wrong. I can add a log message saying that a particular client or server doesn't support the handling of old messages.

Reply:
Added the appropriate logging statement.